ilastik/applets/neuralNetwork/tiktorchController.py

```diff
@@ -102,23 +71,49 @@ def numClasses(self):
         return len(self.knownClasses)
 
 
-@dataclasses.dataclass
+class _MISSING:
+    pass
+
+
+MISSING = _MISSING()
+
+
+@dataclasses.dataclass(frozen=True)
 class BIOModelData:
-    name: str
     # doi, nickname, or path
     modelUri: str
     # model zip
     binary: bytes
     # rdf raw description as a dict
     rawDescription: RawResourceDescription
     # hash of the raw description
-    hashVal: str
+    hashVal: Union[str, _MISSING] = MISSING
```

Comment:
Would `None` work here as a value for uninitialized `hashVal`? Alternatively, we could use an empty string instead of `None` (gives us a better type annotation).

Reply:
yeah, that's simpler -> better :)